internal/manifests/collector/collector.go

```diff
@@ -45,12 +45,14 @@ func Build(params manifests.Params) ([]client.Object, error) {
 	manifestFactories = append(manifestFactories, []manifests.K8sManifestFactory{
 		manifests.FactoryWithoutError(ConfigMap),
 		manifests.FactoryWithoutError(HorizontalPodAutoscaler),
-		manifests.FactoryWithoutError(ServiceAccount),
```

Comment:
Could we instead just check this in the service account manifest? In the builder we would then return nil if service account is set